compiler/rustc_passes/src/check_attr.rs

```diff
@@ -158,6 +158,27 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
                 Attribute::Parsed(AttributeKind::AsPtr(attr_span)) => {
                     self.check_applied_to_fn_or_method(hir_id, *attr_span, span, target)
                 }
+                &Attribute::Parsed(AttributeKind::MayDangle(attr_span)) => {
+                    // Check if `#[may_dangle]` is applied to a lifetime or type generic parameter in `Drop` impl.
```

Comment:
this should be in its own function, not here, just like other validation logic usually isn't